### PR TITLE
Converted ZC_ACK_REQ_CHANGE_MEMBERS to struct

### PR DESCRIPTION
**Pull request by AoShinRO**
* **Addressed Issue(s)**: 

* **Server Mode**: 

* **Description of Pull Request**: Fixed sending of all guildmembers info

### CODE REVIEW

File: src/map/guild.cpp [hunk 1]
```diff
@@ -1493,7 +1493,7 @@ int guild_change_memberposition(int guild_id,uint32 account_id,uint32 char_id,sh
 int guild_memberposition_changed(struct mmo_guild &g,int idx,int pos) {
 
 	g.member[idx].position=pos;
-	clif_guild_memberpositionchanged(g,idx);
+	clif_guild_memberpositionchanged(g);
```
Comment (AoShinRO) [inline src/map/guild.cpp:1496]: maybe we need to change this call location

File: src/map/clif.cpp [hunk 2]
```diff
@@ -8897,23 +8897,28 @@ void clif_guild_positionchanged(const struct mmo_guild &g,int idx)
...
 /// 0156 <packet len>.W { <account id>.L <char id>.L <position id>.L }* (ZC_ACK_REQ_CHANGE_MEMBERS)
-void clif_guild_memberpositionchanged(const struct mmo_guild &g, int idx)
-{
-	// FIXME: This packet is intended to update the clients after a
-	// commit of member position assignment changes, not sending one
-	// packet per position.
-	map_session_data *sd;
-	unsigned char buf[64];
+void clif_guild_memberpositionchanged(const struct mmo_guild &g){
+	map_session_data *sd = guild_getavailablesd(g);
```
Comment (aleos89) [inline src/map/clif.cpp:8901]:
```suggestion
	map_session_data *sd = guild_getavailablesd(g);

```

File: src/map/clif.cpp [hunk 3]
```diff
@@ -8897,23 +8897,28 @@ void clif_guild_positionchanged(const struct mmo_guild &g,int idx)
...
-	WBUFL(buf, 8)=g.member[idx].char_id;
-	WBUFL(buf,12)=g.member[idx].position;
-	// }*
-	if( (sd=guild_getavailablesd(g))!=nullptr )
-		clif_send(buf,WBUFW(buf,2),&sd->bl,GUILD);
+	PACKET_ZC_ACK_REQ_CHANGE_MEMBERS *p = reinterpret_cast<PACKET_ZC_ACK_REQ_CHANGE_MEMBERS*>( packet_buffer );
+
+	p->PacketType = HEADER_ZC_ACK_REQ_CHANGE_MEMBERS;
+	p->PacketLength = sizeof(*p);
+	int c = 0;
```
Comment (aleos89) [inline src/map/clif.cpp:8909]:
```suggestion

	int c = 0;

```

File: src/map/clif.cpp [hunk 4]
```diff
@@ -8897,23 +8897,28 @@ void clif_guild_positionchanged(const struct mmo_guild &g,int idx)
...
+	for(const guild_member& m : g.member){
+		if(m.char_id==0)
```
Comment (aleos89) [inline src/map/clif.cpp:8911]:
```suggestion
		if(m.char_id == 0)
```

File: src/map/clif.cpp [hunk 5]
```diff
@@ -8897,23 +8897,28 @@ void clif_guild_positionchanged(const struct mmo_guild &g,int idx)
...
+			continue;
+		PACKET_ZC_ACK_REQ_CHANGE_MEMBERS_sub& member = p->members[c];
```
Comment (aleos89) [inline src/map/clif.cpp:8913]:
```suggestion

		PACKET_ZC_ACK_REQ_CHANGE_MEMBERS_sub& member = p->members[c];

```

**Review by aleos89 — APPROVED** [review]

**Review by Lemongrass3110 — APPROVED** [review]

**Pull request closed without merging**